modules/services/servers/src/main/java/au/com/shiftyjelly/pocketcasts/servers/di/InterceptorModule.kt

```diff
@@ -1,24 +1,188 @@
 package au.com.shiftyjelly.pocketcasts.servers.di
 
+import au.com.shiftyjelly.pocketcasts.preferences.AccessToken
+import au.com.shiftyjelly.pocketcasts.preferences.Settings
+import au.com.shiftyjelly.pocketcasts.servers.BuildConfig
+import au.com.shiftyjelly.pocketcasts.servers.OkHttpInterceptor
+import au.com.shiftyjelly.pocketcasts.servers.sync.TokenHandler
+import au.com.shiftyjelly.pocketcasts.servers.toClientInterceptor
+import au.com.shiftyjelly.pocketcasts.servers.toNetworkInterceptor
 import com.automattic.android.tracks.crashlogging.CrashLoggingOkHttpInterceptorProvider
 import com.automattic.android.tracks.crashlogging.FormattedUrl
 import com.automattic.android.tracks.crashlogging.RequestFormatter
 import dagger.Module
 import dagger.Provides
 import dagger.hilt.InstallIn
 import dagger.hilt.components.SingletonComponent
+import java.net.HttpURLConnection
+import kotlin.time.Duration.Companion.days
+import kotlin.time.Duration.Companion.minutes
+import kotlinx.coroutines.runBlocking
 import okhttp3.Interceptor
 import okhttp3.Request
+import okhttp3.logging.HttpLoggingInterceptor
 
 @InstallIn(SingletonComponent::class)
 @Module
-class InterceptorModule {
-    @Provides
-    @CrashLoggingInterceptor
-    fun provideMonitoringInterceptor(): Interceptor = CrashLoggingOkHttpInterceptorProvider
+object InterceptorModule {
+    private val sixMonths = 180.days.inWholeSeconds
+    private val fiveMinutes = 180.minutes.inWholeSeconds
+    private val cacheControlHeader = "Cache-Control"
+
+    private val crashLoggingInterceptor = CrashLoggingOkHttpInterceptorProvider
         .createInstance(object : RequestFormatter {
             override fun formatRequestUrl(request: Request): FormattedUrl {
                 return request.url.host.takeIf { it.contains("pocketcasts") } ?: "filtered"
             }
         })
+
+    private val userAgentInterceptor = Interceptor { chain ->
+        val request = chain.request().newBuilder()
+        request.header("User-Agent", Settings.USER_AGENT_POCKETCASTS_SERVER)
+        chain.proceed(request.build())
+    }
+
+    private val cacheControlInterceptor = Interceptor { chain ->
+        val request = chain.request()
+        val originalResponse = chain.proceed(request)
+        var responseBuilder = originalResponse.newBuilder()
+        if (request.url.host == BuildConfig.SERVER_CACHE_HOST) {
+            responseBuilder = responseBuilder.header(cacheControlHeader, "public, max-age=$fiveMinutes")
+        }
+        responseBuilder.build()
+    }
+
+    private val cacheControlNotesInterceptor = Interceptor { chain ->
+        val request = chain.request()
+        val originalResponse = chain.proceed(request)
+        originalResponse.newBuilder().header(cacheControlHeader, "public, max-age=$sixMonths").build()
+    }
```

Comment:
`cacheControlNotesInterceptor` doesn't seem to be used anymore. Can we remove it?

Reply:
Yes, thanks. It was used in an used client. I'll double check to make sure of it.